bin/prover-client/src/proving_ops/btc_ops.rs

```diff
@@ -0,0 +1,120 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use bitcoin::{key::Secp256k1, Block};
+use musig2::secp256k1::SecretKey;
+use rand::{rngs::StdRng, SeedableRng};
+use strata_btcio::rpc::{traits::Reader, BitcoinClient};
+use strata_primitives::{
+    block_credential,
+    buf::Buf32,
+    operator::OperatorPubkeys,
+    params::{OperatorConfig, Params, ProofPublishMode, RollupParams, SyncParams},
+    vk::RollupVerifyingKey,
+};
+use tracing::debug;
+use uuid::Uuid;
+
+use super::ops::ProvingOperations;
+use crate::{
+    errors::{ProvingTaskError, ProvingTaskType},
+    primitives::prover_input::ZKVMInput,
+    task::TaskTracker,
+};
+
+/// Operations required for BTC block proving tasks.
+#[derive(Debug, Clone)]
+pub struct BtcOperations {
+    btc_client: Arc<BitcoinClient>,
+}
+
+impl BtcOperations {
+    /// Creates a new BTC operations instance.
+    pub fn new(btc_client: Arc<BitcoinClient>) -> Self {
+        Self { btc_client }
+    }
+}
+
+#[async_trait]
+impl ProvingOperations for BtcOperations {
+    type Input = (Block, RollupParams);
+    type Params = u64; // params is the block height
+
+    fn block_type(&self) -> ProvingTaskType {
```

Comment:
The method name `block_type` doesn't seem to go well with the type it returns `ProvingTaskType`.